Recrutify/Recrutify.DataAccess/Repositories/UserRepository.cs

```diff
@@ -64,5 +64,40 @@ public Task BulkAddProjectRolesAsync(Guid projectId, IDictionary<Guid, IEnumerab
 
             return GetCollection().BulkWriteAsync(updateModels);
         }
+
+        public Task BulkUpdateProjectRolesAsync(Guid projectId, IDictionary<Guid, IEnumerable<Role>> newUsersRoles, IDictionary<Guid, IEnumerable<Role>> removeUsersRoles, IDictionary<Guid, IEnumerable<Role>> updateUsersRoles)
+        {
+            var binaryProjectId = new BsonBinaryData(projectId, GuidRepresentation.Standard);
+            var arrayFilters = new List<ArrayFilterDefinition>
+            {
+               new BsonDocumentArrayFilterDefinition<ProjectResult>(new BsonDocument("keyValuePair.k", binaryProjectId)),
+            };
+
+            var updateBuilder = Builders<User>.Update;
+            var updateModelsByNewUsers = newUsersRoles.Select(ur => new UpdateOneModel<User>(
+                                                    _filterBuilder.Eq(u => u.Id, ur.Key),
```

Comment:
put _filterBuilder.Eq(u => u.Id, ur.Key) in var userFilter, and use var in all places

Reply:
+